### PR TITLE
Clean all jobsets on delete namespace.

### PR DESCRIPTION
**Pull request by mbobrovskyi**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
To prevent Handling job with no workload error we need to clean jobsets.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

```
2024-06-29T03:29:55.453135835Z	ERROR	jobframework/reconciler.go:384	Handling job with no workload	{"controller": "jobset", "controllerGroup": "jobset.x-k8s.io", "controllerKind": "JobSet", "JobSet": {"name":"job-set","namespace":"multikueue-nrj5n"}, "namespace": "multikueue-nrj5n", "name": "job-set", "reconcileID": "b28cdef5-4081-4672-934d-f206a91336b4", "job": "multikueue-nrj5n/job-set", "gvk": "jobset.x-k8s.io/v1alpha2, Kind=JobSet", "error": "workloads.kueue.x-k8s.io \"jobset-job-set-273af\" is forbidden: unable to create new content in namespace multikueue-nrj5n because it is being terminated"}
sigs.k8s.io/kueue/pkg/controller/jobframework.(*JobReconciler).ReconcileGenericJob
	/workspace/pkg/controller/jobframework/reconciler.go:384
sigs.k8s.io/kueue/pkg/controller/jobframework.(*genericReconciler).Reconcile
	/workspace/pkg/controller/jobframework/reconciler.go:1086
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile
	/workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:119
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:316
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:266
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:227
2024-06-29T03:29:55.453221044Z	ERROR	controller/controller.go:329	Reconciler error	{"controller": "jobset", "controllerGroup": "jobset.x-k8s.io", "controllerKind": "JobSet", "JobSet": {"name":"job-set","namespace":"multikueue-nrj5n"}, "namespace": "multikueue-nrj5n", "name": "job-set", "reconcileID": "b28cdef5-4081-4672-934d-f206a91336b4", "error": "workloads.kueue.x-k8s.io \"jobset-job-set-273af\" is forbidden: unable to create new content in namespace multikueue-nrj5n because it is being terminated"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:329
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:266
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:227
```

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/multikueue/e2e_test.go [hunk 1]
```diff
@@ -165,6 +165,10 @@ var _ = ginkgo.Describe("MultiKueue", func() {
 	})
 
 	ginkgo.AfterEach(func() {
+		gomega.Expect(util.DeleteAllJobsetsInNamespace(ctx, k8sManagerClient, managerNs)).To(gomega.Succeed())
+		gomega.Expect(util.DeleteAllJobsetsInNamespace(ctx, k8sWorker1Client, worker1Ns)).To(gomega.Succeed())
+		gomega.Expect(util.DeleteAllJobsetsInNamespace(ctx, k8sWorker2Client, worker2Ns)).To(gomega.Succeed())
+
 		gomega.Expect(util.DeleteNamespace(ctx, k8sManagerClient, managerNs)).To(gomega.Succeed())
 		gomega.Expect(util.DeleteNamespace(ctx, k8sWorker1Client, worker1Ns)).To(gomega.Succeed())
 		gomega.Expect(util.DeleteNamespace(ctx, k8sWorker2Client, worker2Ns)).To(gomega.Succeed())
```
Comment (trasc) [inline test/e2e/multikueue/e2e_test.go:174]: Can we first check why `util.DeleteNamespace` is not sufficient?

Comment (mbobrovskyi) [reply test/e2e/multikueue/e2e_test.go:174]: In `util.DeleteNamespace` we just have `DeleteAllJobsInNamespace` but don't have `DeleteAllJobsetsInNamespace`

Comment (mbobrovskyi) [reply test/e2e/multikueue/e2e_test.go]: Moved this logic to `DeleteNamespace`.

**Review by trasc — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**